nameable.rb

```diff
@@ -0,0 +1,31 @@
+class Nameable
+  def correct_name
+    raise NotImplementedError, "#{self.class} has not implemented method '#{__method__}'"
+  end
+end
+
+class BaseDecorator < Nameable
+  attr_accessor :nameable
+
+  def initialize(nameable)
+    super()
+    @nameable = nameable
+  end
+
+  def correct_name
+    @nameable.correct_name
+  end
+end
+
+class CapitalizeDecorator < BaseDecorator
+  def correct_name
+    super().upcase
+  end
+end
+
+class TrimmerDecorator < BaseDecorator
+  def correct_name
+    name = super()
+    name.length > 10 ? name[0...10] : name
+  end
```

Comment:
- [ ] Please use the `nameable` variable directly instead of calling the super method. 
  - You can consider using the below code.
  
 ```ruby
def correct_name
      @nameable.correct_name.length > 10 ? @nameable.correct_name[0..9] : @nameable.correct_name
    end
```